### PR TITLE
Add tests for post /air-quality

### PR DESCRIPTION
**Pull request by james-zafar**

### CODE REVIEW

File: src/api/errors/errors.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from src.api.errors.error_response import Error
+
+
+class UnauthorizedError(Error):
+    code = 'missing_or_invalid_api_key'
+    message = 'The API key is missing or not recognized'
+    status_code = 401
```
Comment (DelaneySteve) [inline src/api/errors/errors.py:7]: I might have missed it but does this actually get used anywhere? Or should it be used in the line in the comments in test_post_air_quality

File: src/api/errors/errors.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+
+class ForbiddenError(Error):
+    code = 'action_forbidden'
+    message = 'The specified user does not have permission to perform this action'
+    status_code = 403
```
Comment (DelaneySteve) [inline src/api/errors/errors.py:13]: same as above

File: tests/unit/test_post_air_quality.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        }
+        resp = self.test_client.post('/air-quality', json=airport_data, headers={'X-API-KEY': self.api_key})
+        self.assertEqual(resp.status_code, 201)
+        self.assertCountEqual(['air_quality'], resp.json())
+        self.assertAlmostEqual(resp.json()['air_quality'], 25.5195, 4)
+
+    def test_predict_fails_with_bad_api_key(self) -> None:
+        api_key = 'some_bad_api_key'
+        resp = self.test_client.post('/air-quality', json=self.data, headers={'X-API-KEY': api_key})
+        self.assertEqual(resp.status_code, 401)
```
Comment (DelaneySteve) [inline tests/unit/test_post_air_quality.py:88]: Should this be UnauthorizedError().status_code?

File: tests/unit/test_post_air_quality.py [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        self.assertEqual(resp_json['status_code'], resp.status_code)
+        # There should be exactly one error
+        self.assertEqual(len(resp_json['errors']), 1)
+        # The error should be an `UnauthorizedError`
+        self.assertEqual(resp_json['errors'][0], UnauthorizedError().json())
+
+    @unittest.skip('Response model for missing API key is inconsistent with API schema')
+    def test_predict_fails_with_missing_api_key(self) -> None:
+        resp = self.test_client.post('/air-quality', json=self.data)
+        self.assertEqual(resp.status_code, 403)
```
Comment (DelaneySteve) [inline tests/unit/test_post_air_quality.py:102]: same idea as above

File: src/api/errors/errors.py (hunk 1 above)
Comment (james-zafar) [reply src/api/errors/errors.py:7]: Yes, it's used here: https://github.com/patrickcap/APAQ/pull/34/files#diff-4a8f644b032c2adb9aa1bd80028272b3145771c71962302f44c4d5f9b35e8527R35

**Review by DelaneySteve — APPROVED** [review]

**Pull request merged**